src/Network/Network/NatGateway/GetAzureRMNatGatewayCommand.cs

```diff
@@ -0,0 +1,137 @@
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Nat Gateway Id",
+            ParameterSetName = GetByResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string NatGatewayId { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The resource reference to be expanded.",
+            ParameterSetName = "Expand",
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string ExpandResource { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (this.IsParameterBound(c => c.NatGatewayId))
+            {
+                var resourceIdentifier = new ResourceIdentifier(this.NatGatewayId);
+                this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                this.Name = resourceIdentifier.ResourceName;
+            }
+
+            if (!string.IsNullOrEmpty(this.Name))
+            {
+                var vNatGateway = this.NetworkClient.NetworkManagementClient.NatGateways.Get(ResourceGroupName, Name, ExpandResource);
+                var vNatGatewayModel = NetworkResourceManagerProfile.Mapper.Map<CNM.PSNatGateway>(vNatGateway);
+                vNatGatewayModel.ResourceGroupName = this.ResourceGroupName;
+                vNatGatewayModel.Tag = TagsConversionHelper.CreateTagHashtable(vNatGateway.Tags);
+                WriteObject(vNatGatewayModel, true);
+            }
+            else
+            {
+                IPage<NatGateway> vNatGatewayPage;
+                if(!string.IsNullOrEmpty(this.ResourceGroupName))
+                {
+                    vNatGatewayPage = this.NetworkClient.NetworkManagementClient.NatGateways.List(this.ResourceGroupName);
+                }
+                else
+                {
+                    vNatGatewayPage = this.NetworkClient.NetworkManagementClient.NatGateways.ListAll();
+                }
+
+                var vNatGatewayList = ListNextLink<NatGateway>.GetAllResourcesByPollingNextLink(vNatGatewayPage,
+                    this.NetworkClient.NetworkManagementClient.NatGateways.ListNext);
+                List<PSNatGateway> psNatGatewayList = new List<PSNatGateway>();
+                foreach (var vNatGateway in vNatGatewayList)
+                {
+                    var vNatGatewayModel = NetworkResourceManagerProfile.Mapper.Map<CNM.PSNatGateway>(vNatGateway);
+                    vNatGatewayModel.ResourceGroupName = NetworkBaseCmdlet.GetResourceGroup(vNatGateway.Id);
+                    vNatGatewayModel.Tag = TagsConversionHelper.CreateTagHashtable(vNatGateway.Tags);
+                    psNatGatewayList.Add(vNatGatewayModel);
+                }
+                WriteObject(psNatGatewayList, true);
```

Comment:
AFAIK wildcard support also requires this update:
`WriteObject(TopLevelWildcardFilter(ResourceGroupName, Name, psNatGatewayList), true);`